griptape/drivers/image_generation/base_image_generation_driver.py

```diff
@@ -11,11 +11,32 @@
 class BaseImageGenerationDriver(ExponentialBackoffMixin, ABC):
     model: str = field(kw_only=True)
 
-    def generate_image(self, prompts: list[str], negative_prompts: Optional[list[str]] = None) -> ImageArtifact:
-        for attempt in self.retrying():
-            with attempt:
-                return self.try_generate_image(prompts=prompts, negative_prompts=negative_prompts)
+    @abstractmethod
+    def try_text_to_image(self, prompts: list[str], negative_prompts: Optional[list[str]] = None) -> ImageArtifact:
```

Comment:
What do you think about creating a `run` method that calls the `try_` methods based on some user criteria? I think methods prefixed with "try" are not generally for public use. They should probably be marked as protected, but that's a separate conversation.